File: app/src/main/java/cn/intret/app/picgo/model/event/RecentOpenFolderListChangeMessage.java
package cn.intret.app.picgo.model.event;

import java.util.List;

import cn.intret.app.picgo.model.user.RecentRecord;

public class RecentOpenFolderListChangeMessage extends BaseMessage {
    private List<RecentRecord> mRecentRecord;

    public RecentOpenFolderListChangeMessage(String desc, List<RecentRecord> recentRecord) {
        super(desc);
        mRecentRecord = recentRecord;
    }

    public RecentOpenFolderListChangeMessage() {
    }

    public RecentOpenFolderListChangeMessage(List<RecentRecord> recentRecord) {
        mRecentRecord = recentRecord;
    }

    public RecentOpenFolderListChangeMessage setRecentRecord(List<RecentRecord> recentRecord) {
        mRecentRecord = recentRecord;
        return this;
    }

    public List<RecentRecord> getRecentRecord() {
        return mRecentRecord;
    }

    @Override
    public String toString() {
        return "RecentOpenFolderListChangeMessage{" +
                "mRecentRecord=" + mRecentRecord +
                "} " + super.toString();
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/event/RecentOpenFolderListChangeMessage.kt
package cn.intret.app.picgo.model.event

import cn.intret.app.picgo.model.user.RecentRecord

class RecentOpenFolderListChangeMessage : BaseMessage {
    private var mRecentRecord: List<RecentRecord>? = null

    constructor(desc: String, recentRecord: List<RecentRecord>) : super(desc) {
        mRecentRecord = recentRecord
    }

    constructor()

    constructor(recentRecord: List<RecentRecord>) {
        mRecentRecord = recentRecord
    }

    fun setRecentRecord(recentRecord: List<RecentRecord>): RecentOpenFolderListChangeMessage {
        mRecentRecord = recentRecord
        return this
    }

    fun getRecentRecord(): List<RecentRecord>? {
        return mRecentRecord
    }

    override fun toString(): String {
        return "RecentOpenFolderListChangeMessage{" +
                "mRecentRecord=" + mRecentRecord +
                "} " + super.toString()
    }
}
